beacon-chain/rpc/validator/attester.go

```diff
@@ -84,6 +91,21 @@ func (vs *Server) GetAttestationData(ctx context.Context, req *ethpb.Attestation
 		return nil, status.Errorf(codes.Internal, "Could not retrieve head root: %v", err)
 	}
 
+	// In the case that we receive an attestation request after a newer state/block has been
+	// processed, we walk up the chain until state.Slot <= req.Slot to prevent producing an
+	// attestation that violates processing constraints.
+	for headState.Slot() > req.Slot {
+		if ctx.Err() != nil {
+			return nil, status.Errorf(codes.Aborted, ctx.Err().Error())
+		}
+		parent := headState.ParentRoot()
+		headRoot = parent[:]
+		headState, err = vs.BeaconDB.State(ctx, parent)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
```

Comment:
what if headState is `nil`? we do delete states and even more with the new state gen service

Reply:
Good point. Do we have state gen service? What do you recommend?

I can return an error as an easy resolution, but I would prefer the more viable / profitable solution

Reply:
Something like this:
https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/blockchain/head.go#L124

With also an error check at the end:
```go
if headState == nil {
	return status.Error...
}
```